reviews/views.py

```diff
@@ -0,0 +1,66 @@
+import json, re, bcrypt, jwt
+
+from django.views           import View
+from django.http            import JsonResponse
+from django.db.models       import Avg, Q
+
+from reviews.models         import Review, ReviewImage
+from users.models           import User, WishList
+from users.utils            import login_decorator
+from restaurants.models     import Restaurant
+
+class ReviewView(View):
+    @login_decorator
+    def post(self, request, restaurant_id):
+        data    = json.loads(request.body)
+        user    = request.user
+
+        if Review.objects.filter(user_id = user.id, restaurant_id=restaurant_id).exists():
+            return JsonResponse({'MESSAGE':'REVIEW_EXIST'}, status=400)
+
+        review = Review.objects.create(
+            restaurant_id = restaurant_id,
+            user_id       = user.id,
+            description   = data['description'],
+            rating        = data['rating'],
+        )
+
+        ReviewImage.objects.create(
+            review_id = review.id,
+            image     = data['image'],
+        )
+
+        return JsonResponse({'MESSAGE':'SUCCESS'},status = 201)
+
+    @login_decorator
+    def patch(self, request, restaurant_id, review_id):
+        data = json.loads(request.body)
+        user = request.user
+
+        if not Review.objects.filter(id=review_id, user_id=user.id).exists():
+            return JsonResponse({'MESSAGE':'NOT_EXISTS'}, status=400)
+
+        Review.objects.filter(id=review_id, user_id=user.id).update(
+                restaurant_id = restaurant_id,
+                user_id       = user.id,
+                description   = data['description'],
+                rating        = data['rating'],
+                )
+
+        ReviewImage.objects.filter(review_id = review_id).update(
+            review_id = review_id,
+            image     = data['image'],
+            )
+
+        return JsonResponse({'MESSAGE':'SUCCESS'}, status = 200)
+
+    @login_decorator
+    def delete(self, request, restaurant_id, review_id):
+        user = request.user
+
+        if not Review.objects.filter(id=review_id, user_id=user.id).exists():
+            return JsonResponse({'MESSAGE':'NOT_EXISTS'}, status=400)
+
+        Review.objects.get(id=review_id, user_id=user.id).delete()
+
+        return JsonResponse({"MESSAGE": 'SUCCESS'}, status=204)
```

Comment:
`Review.objects.get(id=review_id, user_id=user.id).delete()` 했을 경우에 매칭되는 데이터가 없는 경우에 대한 에러 핸들링이 안되어 있는 것 같습니다! 추가해주시면 좋을 것 같아요~

```suggestion
        user = request.user

        if not Review.objects.filter(id=review_id, user_id=user.id).exists():
            return JsonResponse({'MESSAGE':'NOT_EXISTS'}, status=400)

        Review.objects.get(id=review_id, user_id=user.id).delete()

        return JsonResponse({"MESSAGE": 'SUCCESS'}, status=204)
```

Reply:
엇..! 61번부터 있는 if문이 존재여부를 확인하고 삭제를 진행하는 코드로 구성하였습니다..ㅎㅎ